### PR TITLE
Add user ID to PersistentStorageInstance (#36)

### PR DESCRIPTION
**Pull request by matthew-richerson**
Store the user ID of the user that created a PersistentStorageInstance. Don't allow another user to successfully create or delete the PersistentStorageInstance.

I also added a common way to set the object.Status.Error field based on the error returned from the reconcile function.
Signed-off-by: Matt Richerson <matthew.richerson@dhcp-10-113-10-52.its.hpecorp.net>

### CODE REVIEW

File: controllers/directivebreakdown_controller.go [hunk 1]
```diff
@@ -299,11 +304,16 @@ func (r *DirectiveBreakdownReconciler) createOrUpdatePersistentStorageInstance(c
 				if err != nil {
 					return err
 				}
+			} else {
+				if psi.Spec.UserID != dbd.Spec.UserID {
+					return dwsv1alpha1.NewResourceError(fmt.Sprintf("Existing persistent storage user ID %v does not match user ID %v", psi.Spec.UserID, dbd.Spec.UserID), nil).WithUserMessage("User ID does not match existing persistent storage").WithFatal()
```
Comment (NateRoiger) [inline controllers/directivebreakdown_controller.go:309]: I'm trying to add `NewResourceErrorf` in PR #70 - consider using it if I can commit ahead of you (or add it yourself)
https://github.com/NearNodeFlash/nnf-sos/blob/fdf9573394a7efbf5dd827543f403fb87235807c/api/v1alpha1/workflow_error.go#L42-L44

Comment (matthew-richerson) [reply controllers/directivebreakdown_controller.go:309]: Yeah I saw that. Looks awesome. I'll switch it once your PR is in.

**Review by NateRoiger — APPROVED** [review]

**Review by ajfloeder — APPROVED** [review]

**Review by bdevcich-hpe — APPROVED** [review]

**Comment by NateRoiger** [comment]
The latest test failure seems very similar to what I was seeing last week

https://github.com/NearNodeFlash/nnf-sos/actions/runs/3106817788/jobs/5034167341#step:8:6373
```
2022-09-22T15:46:06.651Z	INFO	controllers.NnfWorkflow	Finish wait	{"Workflow": "default/jobdw-lustre-8c395fff", "state": "pre_run", "index": 0, "reason": "pending mount", "object": "default/jobdw-lustre-8c395fff-0-computes", "kind": "NnfAccess"}
```

Did the merge pull in my workaround, or do you think there's something more to be done?

**Comment by matthew-richerson** [comment]
Your fix is in there, so there still must be an issue. I wasn't going to worry about it because when I add the changes to update the ResourceError fields when the reconciler returns, I think that will make some of this go away.

**Review by NateRoiger — APPROVED** [review]

**Pull request merged**